app/components/common/fork/common/url.filter.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+var angular = require('angular');
+
+angular.module('common.url', [])
+  .filter('url', function (base_url) {
+    return urlFilter;
+
+    function urlFilter (path, query) {
+      query = query || {};
+
+      // If relative path, prefix it with base_url
+      var url = /^https?:\/\//.test(path) ? path : base_url + '/#!' + path;
+
+      var querystring = Object.keys(query).map(function (param) {
```

Comment:
I believe $httpParamSerializer is what you're looking for here( https://docs.angularjs.org/api/ng/service/$httpParamSerializer)